spec/requests/floating_ips_spec.rb

```diff
@@ -180,4 +182,38 @@
       expect_bad_request(/Delete for Floating Ip/)
     end
   end
+
+  describe 'OPTIONS /api/floating_ips' do
+    it 'returns a DDF schema for add when available via OPTIONS' do
+      zone = FactoryBot.create(:zone, :name => "api_zone")
+      provider = FactoryBot.create(:ems_network, :zone => zone)
+
+      stub_supports(provider.class::FloatingIp, :create)
+      stub_params_for(provider.class::FloatingIp, :create, :fields => [])
+
+      options(api_floating_ips_url(:ems_id => provider.id))
+
+      expect(response.parsed_body['data']).to match("form_schema" => {"fields" => []})
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe 'OPTIONS /api/floating_ips/:id' do
+    it 'returns a DDF schema for edit when available via OPTIONS' do
+      #ems    = FactoryBot.create(:ems_openstack)
+      #tenant = FactoryBot.create(:cloud_tenant_openstack, :ext_management_system => ems)
+      #floating_ip = FactoryBot.create(:floating_ip_openstack, :ext_management_system => ems.network_manager, :cloud_tenant => tenant)
+
+      provider = FactoryBot.create(:ems_openstack, :name => 'test_provider')
```

Comment:
You don't actually need to create a provider object here, you can work with just the floating_ip record